src/main/java/jenkins/plugins/logstash/persistence/BuildData.java

```diff
@@ -152,6 +152,11 @@ public BuildData(AbstractBuild<?, ?> build, Date currentTime) {
         }
       }
     }
+    try {
+      buildVariables.putAll(build.getEnvironment(listener));
+    } catch (Exception e) {
```

Comment:
this looks fishy..
- what kind of exception are you anticipating here? Can we change this to a more specific class?
- can you log the exception at TRACE/FINE level so that it doesn't disappear completely?